### PR TITLE
Fix and modernize Chebyshev(Tensor|Vec)Function

### PR DESCRIPTION
**Pull request by akva2**
This has been on my tech debt list for a while

- add proper tests for Chebyshev functions
- add domain specification for Chebyshev functions
- clean up the derivatives mess. we can now use the standard API for this

### CODE REVIEW

**Comment by akva2** [comment]
jenkins build this with downstreams please

**Review by kmokstad — APPROVED** [review]
Looks fine, although I didn't go into the details.

The only comment is that the commits could be less fragmented.
For instance, the first two commits could be merged into one, and commits 7-10 could be one commit only, so also the last two commit

File: src/Utility/Functions.C [hunk 1]
```diff
@@ -925,10 +910,10 @@ VecFunc* utl::parseVecFunc (const std::string& func, const std::string& type,
     IFEM::cout <<": "<< v;
     return new ConstVecFunc(v);
   }
-  else if (type == "chebyshev")
-    return new ChebyshevVecFunc(splitValue(func),false);
-  else if (type == "chebyshev2")
-    return new ChebyshevVecFunc(splitValue(func),true);
+  else if (type == "chebyshev") {
+    return new ChebyshevVecFunc(splitString(func),true,false);
+  } else if (type == "chebyshev2")
```
[the remark this answers is not in the document]
Comment (akva2) [reply src/Utility/Functions.C]: although it is removed in a later commit i have fixed this.

**Comment by akva2** [comment]
jenkins build this with downstreams ifem-stokes=258 please

**Pull request merged**